pkg/kubecost/summaryallocation.go

```diff
@@ -831,18 +831,32 @@ func (sas *SummaryAllocationSet) AggregateBy(aggregateBy []string, options *Allo
 			// which equals the proportion of filtered-to-actual cost.
 			cpuFilterCoeff := 0.0
 			if allocTotals[key].CPUCost > 0.0 {
-				cpuFilterCoeff = allocTotalsAfterFilters[key].CPUCost / allocTotals[key].CPUCost
+				filteredAlloc, ok := allocTotalsAfterFilters[key]
+				if ok {
+					cpuFilterCoeff = filteredAlloc.CPUCost / allocTotals[key].CPUCost
+				} else {
+					cpuFilterCoeff = 0.0
+				}
```

Comment:
I was unable to create a situation where `allocTotals` had a key for an idle allocation that `allocTotalsAfterFilters` did not. Planning to ask Niko about it when he gets back, because maybe it's possible.

In any case, the initial stack trace we saw lead us to believe that `key` may be present in `allocTotals` but not `allocTotalsAfterFilters`. If that is the case, then I believe the allocation in question would have been filtered out, and the correct action is to set this ratio to 0

Reply:
Happy to talk about this one if you want to. In general, the fix looks good :sunglasses: